recipes/zyre/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+from conans import ConanFile, CMake, tools
+from sys import platform
+import re
+import os
+
+
+class ZyreConan(ConanFile):
+    name = "zyre"
+    version = "2.0.1"
+    license = "MPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/zeromq/zyre"
+    description = "Local Area Clustering for Peer-to-Peer Applications."
+    topics = ("conan", "zyre", "czmq", "zmq", "zeromq",
+              "message-queue", "asynchronous")
+    exports_sources = ['CMakeLists.txt', 'Findzyre.cmake']
+    settings = "os", "compiler", "build_type", "arch"
+    requires = "zeromq/4.3.2", "czmq/4.2.0"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "drafts": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "drafts": False,
+    }
+    generators = ["cmake"]
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.compiler == 'Visual Studio':
+            del self.options.fPIC
+
+    def build_requirements(self):
+        if not tools.which("ninja") and \
+                self.settings.compiler == 'Visual Studio':
+            self.build_requires.add('ninja/1.9.0')
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        generator = 'Ninja' if self.settings.compiler == "Visual Studio" \
+            else None
```

Comment:
Is this absolutely required?
You can always define `CONAN_CMAKE_GENERATOR` ([documentation](https://docs.conan.io/en/latest/reference/env_vars.html#conan-cmake-generator)) in your profile or environment if you absolutely want to use ninja.
ninja is only absolutely required (more or less) when using `meson`.